src/grid/cubic.py

```diff
@@ -0,0 +1,516 @@
+        if l_bnd.nsize != 3 or u_bnd.nsize != 3 or step_sizes.nsize != 3:
+            raise ValueError("The arguments, l_bnd, u_bnd, step_sizes should all have size 3.")
+        if np.any(u_bnd >= l_bnd):
+            raise ValueError("In each coordinate, u_bnd should be greater than l_bnd.")
+        if np.any(step_sizes <= 0):
+            raise ValueError("In each coordinate, step_sizes should be positive.")
+
+        r"""
+        Input is either
+        
+        (oned_Grid, oned_grid, oned_grid), Here only points on each direction is only stored.
+        Have a function that returns the poitns if it is needed, i.e. utilizing vectorization
+            for evaluating a function.
+        
+        or 
+        
+        oned_grid, dim=3 so that it results in (oned_grid, oned_grid, oned_grid).
+        Here only points on each direction is only stored.  
+        Have a function that returns the poitns if it is needed, i.e. utilizing vectorization
+            for evaluating a function.
+        
+        or 
+        
+        Grid isn't stored at all, only l_bnd, u_bnd and step_size.
+        Have a function that returns the poitns if it is needed, i.e. utilizing vectorization
+            for evaluating a function.
+        
+        """
+
+        # compute points in each direction/dimension
+        points_1 = np.arange(l_bnd[0], u_bnd[0] + step_sizes[0], step_sizes[0])
+        points_2 = np.arange(l_bnd[1], u_bnd[1] + step_sizes[1], step_sizes[1])
+        points_3 = np.arange(l_bnd[2], u_bnd[2] + step_sizes[2], step_sizes[2])
+        npts_1 = points_1.nsize
+        npts_2 = points_2.nsize
+        npts_3 = points_3.nsize
+        self._num_pts = (npts_1, npts_2, npts_3)
+        points = np.zeros((npts_1 * npts_2 * npts_3, 3))
+        # assign x, y & z coordinates
+        points[:, 0] = np.repeat(points_1, npts_2 * npts_3)
+        points[:, 1] = np.tile(np.repeat(points_2, npts_1), npts_3)
+        points[:, 2] = np.tile(points_3, npts_1 * npts_2)
+        self._points = np.array(points)
+        self._step_sizes = step_size
+
+    @property
+    def points(self):
+        """Return cubic grid points."""
+        return self._points
+
+    @property
+    def step_sizes(self):
+        """Return the step size between to consecutive points."""
+        return self._step_sizes
+
+    @property
+    def num_pts(self):
+        r"""Return the number of points (N_x, N_y, N_z) in each direction of the Cubic Grid."""
+        return self._num_pts
+
+    @property
+    def origin(self):
+        r"""Return the origin of the coordinate system."""
+        # Bottom, Left-Most, Down-most point of the Cubic Grid.
+        return self._points[0]
+
+    def __len__(self):
+        """Return the number of grid points."""
+        return self._points.nsize
+
+    def evaluate_function_on_grid(self, f):
+        r"""
+        Evaluates the function f on each point of the cubic grid.
+
```

Comment:
Can you please explain why this method is needed @Ali-Tehrani? I am just thinking that we don't have any methods of this kind for other grids, and the points are always passed to a function for evaluation on the grid.

Reply:
This is old, I removed it after our meeting last week, resolved in d958179.